### PR TITLE
fix(migration) up part of regex route

### PR DESCRIPTION
**Pull request by Suika-Kong**
Fix FT-3254

### CODE REVIEW

File: kong/db/schema/entities/routes.lua [hunk 1]
```diff
@@ -87,6 +87,8 @@ else
       { methods        = typedefs.methods },
       { hosts          = typedefs.hosts },
       { paths          = typedefs.router_paths },
+      -- default to true so user added routes will be in new format
+      { migrate_3      = { type = "boolean", required = true, default = true }, },
```
Comment (mayocream) [inline kong/db/schema/entities/routes.lua:91]: How about `is_new_regex_format`?

File: kong/db/migrations/core/016_280_to_300.lua [hunk 2]
```diff
@@ -393,6 +401,8 @@ return {
 
       ALTER TABLE routes ADD expression text;
       ALTER TABLE routes ADD priority int;
+
+      ALTER TABLE routes ADD COLUMN migrate_3 boolean DEFAULT FALSE;
```
Comment (mayocream) [inline kong/db/migrations/core/016_280_to_300.lua:405]:
```suggestion
      ALTER TABLE routes ADD migrate_3 boolean DEFAULT FALSE;
```

File: kong/router/atc_compat.lua [hunk 3]
```diff
@@ -14,6 +14,7 @@ local tb_new = require("table.new")
 local tb_clear = require("table.clear")
 local tb_nkeys = require("table.nkeys")
 local yield = require("kong.tools.utils").yield
+local migrate_route = require "kong.db.migrations.migrate_regex_280_300".migrate_route
```
Comment (chronolaw) [inline kong/router/atc_compat.lua:17]: Let's use style `require("xxx").xxx`

Comment (Suika-Kong) [inline kong/router/atc_compat.lua:17]:
```suggestion
local migrate_route = require("kong.db.migrations.migrate_regex_280_300").migrate_route
```

File: kong/db/migrations/migrate_regex_280_300.lua [hunk 4]
```diff
@@ -83,4 +83,17 @@ local function migrate_regex(reg)
   return migrated, true
 end
 
-return migrate_regex
+local function migrate_route(route)
```
Comment (chronolaw) [inline kong/db/migrations/migrate_regex_280_300.lua:86]: how about `migrate_paths`

Comment (Suika-Kong) [reply kong/db/migrations/migrate_regex_280_300.lua:86]: It works on the whole route.

File: kong/db/schema/entities/routes.lua (hunk 1 above)
Comment (Suika-Kong) [reply kong/db/schema/entities/routes.lua]: done

**Comment by dndx** [comment]
Should `is_new_regex_format` be dropped in the teardown phase? Will this field be shown to the user when they request Admin APIs?

**Comment by bungle** [comment]
Things that worry me:
- on up we do both schema changes and data migrations, this has been especially flaky with cassandra clusters
- we add a new visible property to schema, and thus everywhere, including admin api and daos. Can this property be mistakenly set to wrong value

**Comment by Suika-Kong** [comment]
> Should `is_new_regex_format` be dropped in the teardown phase? Will this field be shown to the user when they request Admin APIs?

That will fail its purpose to make it  safe to do this migration multiple times.

**Comment by Suika-Kong** [comment]
Closing because we decide to do it another way.

**Pull request closed without merging**